book/style_guide.md

```diff
@@ -0,0 +1,221 @@
+# Best practices
+
+This page lists our best practices we advice to follow. Keep in mind that they are not required to be
+used in external repositories (not ours), you can change them in the way you want,
+but please be consistent and follow your rules.
+
+## Formatting
+
+### Defaults
+
+**It's recommended to** assume that by default no spaces or tabs allowed, but the following rules define where they are allowed.
+
+### Basic
+
+- **It's recommended to** put one space before and after pipe `|` symbol, commands, subcommands, their options and arguments.
+- **It's recommended to** omit commas between list items.
+
+Correct:
+
+```nushell
+'Hello, Nushell! This is a gradient.' | ansi gradient --fgstart '0x40c9ff' --fgend '0xe81cff'
+```
+
+Incorrect:
+
+```nushell
+'Hello, Nushell! This is a gradient.' |  ansi gradient --fgstart '0x40c9ff' --fgend '0xe81cff' # two many spaces after "|": 2 instead of 1
+```
+
+#### One-line format
+
+One-line format is a format for writing all commands in one line.
+
+1. parameters:
+   1. **It's recommended to** put no spaces before and after pipe `|` symbol denoting block or closure parameter list beginning.
+   2. **It's recommended to** put one space after comma `,` after block or closure parameter parameter.
+   3. **It's recommended to** put no space before and one space after pipe `|` symbol denoting block or closure parameter list end.
+2. block and closure bodies:
+   1. **It's recommended to** put one space before closing block or closure curly brace `}`.
+3. records:
+   1. **It's recommended to** put one space before first record key and after last record key value.
```

Comment:
i'm not sure about that :thinking: 

i tend to write
```
if $conf { foo } else { bar }
```
but
```
let rec = {a: 1, b: 2}
```
looks good to me :yum: 

to me, the block `{ ... }` makes it clear the inner part is the content of the block but the `{}` are just syntax.
on the other hand, `{key: value}` makes the record look more like one block of data.

Reply:
as i read it now, it's like with lists to me and i never write `[ 1 2 3 ]` :open_mouth:

Reply:
> i never write [ 1 2 3 ] open_mouth

Agree. 😸

Reply:
Done. ;)

Reply:
noishe :ok_hand: